### PR TITLE
[docs] Add a section on Rust debugging to the Rust guide

### PR DESCRIPTION
**Pull request by JesterOrNot**
See #552 :smiley:

### CODE REVIEW

File: src/docs/languages/rust.md [hunk 1]
```diff
@@ -84,12 +84,144 @@ vscode:
...
+vscode:
+  extensions:
+    - webfreak.debug@0.24.0:1zVcRsAhewYEX3/A9xjMNw==
+```
+
+If you already added a extension just add the following below your other extensions.
+
+```yaml
+    - webfreak.debug@0.24.0:1zVcRsAhewYEX3/A9xjMNw==
+```
```
Comment (nisarhassan12) [inline src/docs/languages/rust.md:113]: I think there is no need for differentiating if you have not added an extension then do x if you did maybe just keep it consistent like it is in the above sections. I think this is already very self-explanatory 
image

Maybe rephrase this to something like the following:

```suggestion
First, before we get to that we need to get some prerequisites set-up first we'll add the [Native Debug](https://github.com/WebFreak001/code-debug) extension for adding it just put the following snippet in your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/) file:

```yaml
vscode:
  extensions:
    - webfreak.debug@0.24.0:1zVcRsAhewYEX3/A9xjMNw==
```

Comment (nisarhassan12) [reply src/docs/languages/rust.md:113]: I think it's ok to have it @JesterOrNot please ignore the above comment. I'll mark it as resolved.

File: src/docs/languages/rust.md [hunk 2]
```diff
@@ -84,12 +84,144 @@ vscode:
...
     - bungcip.better-toml@0.3.2:3QfgGxxYtGHfJKQU7H0nEw==
 ```
 
+## Debugging
+
+In this section we will show you how to configure your project for debugging in Gitpod.
+
+First, before we get to that we need to get some prerequisites set-up first we'll add the needed extension
+
+If you havn't added extensions to your repoository already add the following snippet to your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/) file:
```
Comment (nisarhassan12) [inline src/docs/languages/rust.md:101]:
```suggestion
If you haven't added extensions to your repository already add the following snippet to your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/) file:
```

Comment (nisarhassan12) [reply src/docs/languages/rust.md:101]: Also maybe it would be better to reference the name of the extension here.

File: src/docs/languages/rust.md [hunk 3]
```diff
@@ -84,12 +84,144 @@ vscode:
...
+image:
+  file: .gitpod.Dockerfile
+```
+
+Now that thats out of the way, here is a video on how to configure the debug configuration
+
+![Rust debugging example](../images/RustDebug.gif)
+
+So, basically in this video we:
+1. Then, go to the debug menu and select "Add Configuration..."
```
Comment (nisarhassan12) [inline src/docs/languages/rust.md:157]: I'm not good at English but is this really supposed to start with a "Then" I was not expecting it to be there.

 ```suggestion
1. Go to the debug menu and select "Add Configuration..."
```

File: src/docs/languages/rust.md (hunk 2 above)
Comment (JesterOrNot) [reply src/docs/languages/rust.md]: It is any extension at all this is to prevent the accidental creation of duplicate keys
